src/ctia/stores/es/crud.clj

```diff
@@ -316,51 +316,61 @@ It returns the documents with full hits meta data including the real index in wh
                                            ident
                                            get-in-config))))))
 
-(s/defn make-search-query
-  [{{:keys [default_operator]} :props
+(s/defn make-search-query :- {s/Keyword s/Any}
+  "Translate SearchQuery map into ES Query DSL map"
+  [{{:keys [default_operator]}               :props
     {{:keys [get-in-config]} :ConfigService} :services} :- ESConnState
-   {:keys [query-string filter-map range]} :- SearchQuery
+   {:keys [filter-map
+           range
+           full-text]} :- SearchQuery
    ident]
-  (let [es-query-string {:query_string (into {:query query-string}
-                                             (when default_operator
-                                               {:default_operator default_operator}))}
-        date-range-query (when range
-                           {:range range})
+  (let [range-query  (when range
+                       {:range range})
         filter-terms (-> (ensure-document-id-in-map filter-map)
-                         q/prepare-terms)]
+                            q/prepare-terms)
+        full-text-q  (when full-text
+                       {(get full-text :query_mode :query_string) ;; if no :query-mode passed, use :query_string mode
+                        (merge
+                         (dissoc full-text :query_mode)
+                         (when (and default_operator
+                                    ;; multi_match queries in ES don't support default_operator
+                                    (not= :multi_match (:query_mode full-text)))
```

Comment:
Ok, like I said, I let it to your appreciation.
Just in case, here is a proposal
```suggestion
        es_query (:query_mode full-text :query_string) ;; if no :query-mode passed, use :query_string mode
        full-text-q  (when full-text
                       {es_query
                        (merge
                         (dissoc full-text :query_mode)
                         (when (and default_operator
                                    ;; multi_match queries in ES don't support default_operator
                                    (contains? #{:simple_query_string :query_string} es_query))
```